Source/WebCore/accessibility/AccessibilityObject.cpp

```diff
@@ -404,6 +404,24 @@ std::optional<SimpleRange> AccessibilityObject::misspellingRange(const SimpleRan
     return std::nullopt;
 }
 
+std::optional<CharacterRange> AccessibilityObject::textInputMarkedRange() const
+{
+    auto node = this->node();
+    if (!node)
+        return std::nullopt;
+
+    auto* frame = node->document().frame();
+    if (!frame)
+        return std::nullopt;
+
+    auto range = frame->editor().compositionRange();
+    if (!range)
+        return std::nullopt;
+
+    auto scope = makeRangeSelectingNodeContents(*node);
+    return characterRange(scope, *range);
```

Comment:
nit: You can get away `return { { scope, *range } };`

Reply:
I'll update it to reflect that, thanks

Reply:
Unfortunately, I think that this can't be written with list initialization syntax, since technically this function is not a constructor. It's declared at the top of `TextIterator.h` and is designed for converting between `SimpleRange` and `CharacterRange` types. 

If there's something I'm missing that would make this possible though, please let me know! I couldn't get it to compile with this suggested change.

Reply:
My bad! I assumed you were calling a constructor. This is fine as-is then

Reply:
All good! There are still many things I have to learn about C++ so these suggestions are always helpful even if they don't pan out.